ddl/partition.go

```diff
@@ -1628,6 +1655,37 @@ func checkExchangePartitionRecordValidation(w *worker, pt *model.TableInfo, inde
 	return nil
 }
 
+func checkExchangePartitionPlacementPolicy(t *meta.Meta, ntPlacementPolicyRef *model.PolicyRefInfo, ptPlacementPolicyRef *model.PolicyRefInfo) (bool, error) {
+	if ntPlacementPolicyRef == nil && ptPlacementPolicyRef == nil {
+		return true, nil
+	}
+	if ntPlacementPolicyRef == nil || ptPlacementPolicyRef == nil {
+		return false, dbterror.ErrPlacementPolicyNotEqual
+	}
+
+	ptPlacementPolicyInfo, _ := getPolicyInfo(t, ptPlacementPolicyRef.ID)
+	ntPlacementPolicyInfo, _ := getPolicyInfo(t, ntPlacementPolicyRef.ID)
+
+	if ntPlacementPolicyInfo == nil && ptPlacementPolicyInfo == nil {
+		return true, nil
+	}
+	if ntPlacementPolicyInfo == nil || ptPlacementPolicyInfo == nil {
+		return false, dbterror.ErrPlacementPolicyNotEqual
+	}
+
+	ptSetting, ntSetting := ntPlacementPolicyInfo.PlacementSettings, ptPlacementPolicyInfo.PlacementSettings
+
+	if ptSetting.PrimaryRegion != ntSetting.PrimaryRegion || ptSetting.Regions != ntSetting.Regions ||
```

Comment:
I think we could use the name of placement policy info to check if they're equal, it doesn't give obvious benefit allowing the exchange when we have 2 policies that are 'deep equal'.

BTW, if you want to check if 2 variables are 'deep-equaled', you could use [`DeepEqual`](https://pkg.go.dev/reflect#DeepEqual) to check all the fields for 2 "PlacementSettings".

Reply:
done.